common/src/main/proto/TransportMessages.proto

```diff
@@ -610,3 +612,24 @@ message PbTransportableError {
   int64 streamId = 1;
   string message = 2;
 }
+
+enum PbAuthType {
+  CLIENT_AUTH = 0;
+  CONNECTION_AUTH = 1;
+}
+
+message PbSaslMechanism {
+  string mechanism = 1;
+  repeated PbAuthType authTypes = 2;
+}
+
+message PbSaslRequest {
+  string appId = 1;
+  string method = 2;
+  PbAuthType authType = 3;
+  bytes payload = 4;
+}
+
+message PbSaslResponse {
+  bytes payload = 1;
```

Comment:
Add sasl status ?

Reply:
It's not need in the current implementation. Please refer https://github.com/apache/incubator-celeborn/pull/2105/files#diff-b0e7ad024188c6b0ba98c0a2f0668bafb28c1d85fd8e9907531e219b1754ed0f

If we need it later, we can add it since extending the message with protobuf doesn't create any incompatibility.

Reply:
Agree, this is encapsulted in the payload anyway - and any other failure follows the standard error paths.
Let us drop it from here, and from the design doc as well; thanks for checking this @otterc !